File: 2-add-two-numbers/add-two-numbers.py
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry=0
        dummy=ListNode()
        prev=dummy
        while l1!=None or l2!=None or carry!=0:
            newnode=ListNode()
            prev.next=newnode
            prev=newnode
            if l1!=None and l2!=None:
                sum=l1.val+l2.val+carry
                newnode.val=sum%10
                carry=sum//10
                l1=l1.next
                l2=l2.next
            elif l1==None and l2!=None:
                sum=l2.val+carry
                newnode.val=sum%10
                carry=sum//10
                l2=l2.next
            elif l1!=None and l2==None:
                sum=l1.val+carry
                newnode.val=sum%10
                carry=sum//10
                l1=l1.next
            elif l1==None and l2==None:
                sum=carry
                newnode.val=sum%10
                carry=sum//10
        return dummy.next

File: 2-add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        
       ListNode head=null,tail=null;
       int digit1,digit2,carry=0;
       
       while(l1!=null || l2!=null || carry==1){
          digit1=(l1!=null)?l1.val:0;
          digit2=(l2!=null)?l2.val:0;

          int sum=digit1+digit2+carry;
          ListNode newnode=new ListNode(sum%10);
          if(head==null){
            head=newnode;
            tail=newnode;
          }
          else{
            tail.next=newnode;
            tail=newnode;
          }
          carry=(sum>=10)?1:0;

          if(l1!=null){
            l1=l1.next;
          }
          if(l2!=null){
            l2=l2.next;
          }
       }

       return head;


    }
}